src/services/ghost/GhostServiceManager.ts

```diff
@@ -115,6 +128,101 @@ export class GhostServiceManager {
 		await this.load()
 	}
 
+	/**
+	 * Check if autocomplete is currently snoozed
+	 */
+	public isSnoozed(): boolean {
+		const snoozeUntil = this.settings?.snoozeUntil
+		if (!snoozeUntil) return false
+		return Date.now() < snoozeUntil
+	}
+
+	/**
+	 * Get remaining snooze time in seconds
+	 */
+	public getSnoozeRemainingSeconds(): number {
+		const snoozeUntil = this.settings?.snoozeUntil
+		if (!snoozeUntil) return 0
+		const remaining = Math.max(0, Math.ceil((snoozeUntil - Date.now()) / 1000))
+		return remaining
+	}
+
+	/**
+	 * Snooze autocomplete for a specified number of seconds
+	 */
+	public async snooze(seconds: number): Promise<void> {
+		if (this.snoozeTimer) {
+			clearTimeout(this.snoozeTimer)
+			this.snoozeTimer = null
+		}
+
+		const snoozeUntil = Date.now() + seconds * 1000
+		const settings = ContextProxy.instance.getGlobalState("ghostServiceSettings") ?? {}
+		await ContextProxy.instance.setValues({
+			ghostServiceSettings: {
+				...settings,
+				snoozeUntil,
+			},
+		})
+
+		this.snoozeTimer = setTimeout(() => {
+			void this.unsnooze()
+		}, seconds * 1000)
+
```

Comment:
So we both have a timer, AND a stored deadline value.  That feels duplicate; why both?

Reply:
no, we disable the whole service when it's snoozed. Earlier we had trouble with people having autocomplete running when it shouldnt be so we enable/disable the exact same actual enabling/disabling does